File: easy/219. Contains Duplicate II/219. Contains Duplicate II.ts
function containsNearbyDuplicate(nums: number[], k: number): boolean {
    const seen = new Map()

    for (let i = 0; i <= nums.length; i++) {
        if (seen.has(nums[i])) {
            const j = seen.get(nums[i])
            if (Math.abs(i - j) <= k)
                return true
        }
        seen.set(nums[i], i)
    }

    return false
};

File: easy/219. Contains Duplicate II/219. Contains Duplicate II.py
from typing import Dict, List


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        seen: Dict[int, int] = {}

        for i in range(len(nums)):
            if nums[i] in seen:
                if abs(seen[nums[i]] - i) <= k:
                    return True
            seen[nums[i]] = i

        return False
